test/Microsoft.AspNet.Mvc.Core.Test/ParameterBinding/ControllerActionArgumentBinderTests.cs

```diff
@@ -400,6 +438,9 @@ private class TestController
             [ValueProviderMetadata]
             public string ValueBinderMarkedProperty { get; set; }
 
+            [CustomBindingSource]
```

Comment:
How does this come into play? It looks like your test doesn't pass it to `BindingInfo`.